src/app/clusters/service-area-server/service-area-delegate.h

```diff
@@ -0,0 +1,386 @@
+     * @param[out] listIndex the map's index in the list, if found.
+     * @param[out] aSupportedMap copy of the location contents, if found.
+     * @return true if a supported map is found, false otherwise.
+     *
+     * @note may be overloaded in device implementation for optimization, if desired.
+     */
+    virtual bool GetSupportedMapById(uint8_t aMapId, uint32_t & listIndex, MapStructureWrapper & aSupportedMap);
+
+    /**
+     * This method is called by the server instance to add a new map to the list.
+     * The server instance will ensure that the newMap is a valid, unique map.
+     * @param[in] newMap The new map to add.
+     * @param[out] listIndex filled with the list index of the new map, if successful.
+     * @return true if successful, false otherwise.
+     *
+     * @note this function SHOULD double check that the added map won't exceed the maximum list size
+     */
+    virtual bool AddSupportedMap(const MapStructureWrapper & newMap, uint32_t & listIndex) = 0;
+
+    /**
+     * This method is called by the server instance to modify an existing map in the list.
+     * The server instance will ensure that the modifiedMap is a valid, unique map.
+     * @param listIndex[in] The index of the map being modified.
+     * @param modifiedMap[in] A map with the modified contents.
+     * @return true if successful, false otherwise.
+     *
+     * @note this function SHOULD double check that modifiedMap's mapID matches the object at listIndex.
+     */
+    virtual bool ModifySupportedMap(uint32_t listIndex, const MapStructureWrapper & modifiedMap) = 0;
+
+    /**
+     * @return true if supported maps was not already null, false otherwise.
+     */
+    virtual bool ClearSupportedMaps() = 0;
+
+    //*************************************************************************
+    // Selected Locations accessors
+
+    virtual uint32_t GetNumberOfSelectedLocations() = 0;
+
+    /**
+     * @brief Get a selected location using the position in the list.
+     * @param[in] listIndex the position in the list.
+     * @param[in] selectedLocation the selected location value, if found.
+     * @return true if a selected location is found, false otherwise.
+     */
+    virtual bool GetSelectedLocationByIndex(uint32_t listIndex, uint32_t & selectedLocation) = 0;
+
+    /**
+     * @return true if the aLocationId locationID is found in the SelectedLocations list, false otherwise.
+     *
+     * @note may be overloaded in device implementation for optimization, if desired.
+     */
+    virtual bool IsSelectedLocation(uint32_t aLocationId);
+
+    /**
+     * This method is called by the server instance to add a new selected location to the list.
+     * The server instance will ensure that the aLocationId references a SUPPORTED location, and is unique within selected
+     * locations.
+     * @param[in] aLocationId The new locationID to add.
+     * @param[in] listIndex filled with the list index of the new location, if successful.
```

Comment:
This is clearly not an in param....

This mistake seems to be present in a bunch of these comments; please fix them all.

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/efafc7c11f9c4920a385ab2956b662aaa73351d3).